src/main/java/com/redhat/coolstore/service/CatalogService.java

```diff
@@ -39,10 +40,11 @@ public CatalogItemEntity getCatalogItemById(String itemId) {
     }
 
     public void updateInventoryItems(String itemId, int deducts) {
-        InventoryEntity inventoryEntity = getCatalogItemById(itemId).getInventory();
+        CatalogItemEntity catalogItem = em.find(CatalogItemEntity.class, itemId);
+        InventoryEntity inventoryEntity = catalogItem.getInventory();
```

Comment:
This is not necessary but quite curious, I guess it has done some "clean code" refactoring to separate the statement into two different statements. Not needed but interesting.